.github/workflows/zephyr-hal.yml

```diff
@@ -66,11 +65,18 @@ jobs:
           # directory that contains the 'msdk' and 'hal_adi' folders.
           bash ./msdk/.github/workflows/scripts/zephyr-hal.sh
 
-      - name: Push hal_adi repository
-        uses: EndBug/add-and-commit@v9.0.1
-        with:
-          add: '*'
-          message: 'Updating Zephyr MSDK Hal based on msdk PR# $PR_NUM'
-          push: true
-          # Specify path in workspace to the repo you want to push to
-          cwd: './hal_adi'
+      - name: Set up SSH key
+        run: |
+          mkdir -p ~/.ssh
+          echo "${{ secrets.MSDK_SSH_KEY }}" > ~/.ssh/id_rsa
+          chmod 600 ~/.ssh/id_rsa
+
+      - name: Push changes to hal_adi repository
+        run: |
+          echo "$(pwd) - $(ls)"
+          cd ./hal_adi
+          git config --global user.email "actions@github.com"
+          git config --global user.name "GitHub Actions"
+          git add *
+          git commit -m "Update Zephyr MSDK Hal based on MSDK PR# ${PR_NUM}"
```

Comment:
Could you also add
`https://github.com/analogdevicesinc/msdk/pull/${PR_NUM}`
in commit message too, to easily access related MSDK PR.

Reply:
Good idea. GitHub should also automatically reformat the PR link to `#123`, for example, as well.